MRIFieldmaps/src/estimation.jl

```diff
@@ -0,0 +1,107 @@
+"""
+ml_cost(x::Matrix{T},y::Matrix{Complex{T}},z::Matrix{Complex{T}}, β)
+Calculates the regularized ML estimator cost between an estimated phase map and multi-echo scan data
+# Arguments
+* `x::Matrix{T}` - fieldmap estimate (in radians)
+* `y::Matrix{Complex{T}}` - Complex first-echo image data
+* `z::Matrix{Complex{T}}` - Complex second-echo image data
+* `m::Matrix{T}` - normalized weighting data (m ∈ [0,1]:= abs.(conj.(y).*z)./maximum(abs.(conj.(y).*z))), precomputed for speed
+* `β` - Regularization parameter controlling roughness penalty
+"""
+function ml_cost(x::Matrix{T},y::Matrix{Complex{T}},z::Matrix{Complex{T}},m::Matrix{T}, β) where T
+
+    Ψ = (sum(m.* (1 .- cos.(angle.(z) .- angle.(y) .- x)),dims=[1,2]) + β * R(x))[1]
+
+end
+
+"""
+R(x::Matrix{T})
+Regularization term which penalizes roughness in a first-order finite-difference sense
+# Arguments
+* `x::Matrix{T}` - fieldmap estimate (in radians)
+"""
+function R(x::Matrix{T}) where T
+
+    return T.(0.5) * sum(abs2,diff(x,dims=1),dims=[1,2]) + T.(0.5) * sum(abs2,diff(x,dims=2),dims=[1,2])
```

Comment:
```suggestion
    return T(0.5) * sum(abs2, diff(x,dims=1)) + T(0.5) * sum(abs2, diff(x,dims=2))
```
`sum` without `dims` arg will sum it all which is what we want, right?

Reply:
extraneous `dims` also in `ml_cost`, will change.

Reply:
fixed in latest commit